### ISSUE TITLE
Support GetFileInfo for directories

### ISSUE THREAD

**Issue by aspnet-hello** [aspnet/Home#2537]
_From @stilettk on Friday, November 3, 2017 2:56:28 AM_

Currently `GetFileInfo` supports only files: when called on a path that points to a directory it returns `PhysicalFileInfo` which has `IsDirectory == false`. 

So, when given a path, how to decide what to do: to call `GetFileInfo` (if it's a file) or call `GetDirectoryContents` (if it's a folder)? Currently the only way I found isn't very intuitive:
`bool isDirectory = fileProvider.GetDirectoryContents(myPath).Exists`

My suggestion:
`new PhysicalFileProvider(myPath).GetFileInfo("data")` 
Expected:
`PhysicalDirectoryInfo`
Actual:
`PhysicalFileInfo`

_Copied from original issue: aspnet/FileSystem#298_

**Comment by stilettk** [comment]
Hello!
What's the resolution? Is it works as designed?

**Comment by muratg** [comment]
@stilettk Do you have any specific scenario broken?

**Comment by stilettk** [comment]
@muratg No, it was just a suggestion. I was just confused because no label was assigned. Anyway, I think I got it now: "no further action is planned for this issue" seems to refer to "won't fix".

**Comment by muratg** [comment]
@stilettk Ah, got it :+1:

**Comment by Tratcher** [comment]
I've also run into this while working on https://github.com/aspnet/AuthSamples/issues/28. It's not blocking but it is confusing and inconvenient.
